### PR TITLE
x-pack/filebeat/module/salesforce: Remove dashboards

### PR DESCRIPTION
**Pull request by kush-elastic**
## Description
This PR aims to remove dashboards from the recently revamped Salesforce module. The plan is to discontinue support for dashboards created in version `7.15.0`, as the overall Beats framework has ceased support for version `8.X.X` dashboards.

For the newer dashboards, support will continue in the upcoming revamp of the Salesforce Integration, ensuring that the latest features and functionalities are available for users.

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues
- Relates https://github.com/elastic/obs-infraobs-team/issues/1277
- Relates https://github.com/elastic/obs-infraobs-team/issues/1248

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -43,7 +43,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Removed deprecated Sophos UTM from Beats. Use the https://docs.elastic.co/integrations/sophos[Sophos] Elastic integration instead. {pull}38037[38037]
 - Introduce input/netmetrics and refactor netflow input metrics {pull}38055[38055]
 - Update Salesforce module to use new Salesforce input. {pull}37509[37509]
-
+- Remove Dashboards from Revamped Salesforce module. {pull}39028[39028]
```
Comment (shmsr) [inline CHANGELOG.next.asciidoc:46]:
```suggestion
- Remove dashboards from the revamped Salesforce module. {pull}39028[39028]
```

**Review by shmsr — CHANGES_REQUESTED** [review]

**Review by shmsr — APPROVED** [review]

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -43,7 +43,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Removed deprecated Sophos UTM from Beats. Use the https://docs.elastic.co/integrations/sophos[Sophos] Elastic integration instead. {pull}38037[38037]
 - Introduce input/netmetrics and refactor netflow input metrics {pull}38055[38055]
 - Update Salesforce module to use new Salesforce input. {pull}37509[37509]
-
+- Remove dashboards from the revamped Salesforce module. {pull}39028[39028]
```
Comment (shmsr) [inline CHANGELOG.next.asciidoc:46]: We do not need this. Ultimately one changelog goes i.e., for module. No need for this?

**Review by shmsr — APPROVED** [review]
Let's remove the changelog and the rest looks good.

**Comment by kush-elastic** [comment]
Hey @tommyers-elastic,
We need your approval on this PR as part of removing dashboards from Salesforce module.

**Review by tommyers-elastic — APPROVED** [review]

**Pull request merged**